Generator/FixtureGenerator.php

```diff
@@ -223,7 +237,15 @@ public function generate(array $config, $outputDir, ProgressHelper $progress, ar
         }
 
         if (isset($config['entities']['group_types'])) {
-            $this->groupTypeGenerator->generate([], $outputDir, $progress);
+            $groupTypes = $this->groupTypeGenerator->generate([], $outputDir, $progress);
+        }
+
+        if (isset($config['entities']['variant_groups'])) {
+            $variantGroupConfig = $config['entities']['variant_groups'];
+            $this->variantGroupGenerator->setAttributes($attributes);
+            $this->variantGroupGenerator->setLocales($locales);
+            $this->variantGroupGenerator->setGroupTypes($groupTypes);
```

Comment:
i don't like too much the setAttributes, setLocales, setGroupTypes but i understand why you do so, because the service is already instanciated and injected.

maybe a configure($attributes, $locales, $groupTypes) could more explicit and less fragile, by raising an exception in the generate when these values have not been provided.

imho, generally, these setters causes issues later, we should never have setters, if we instanciated the service here, with simple OOP, we would prefer pass these args in the constructor.